src/jit/codegenxarch.cpp

```diff
@@ -4989,6 +4989,20 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
 #endif // defined(_TARGET_X86_)
 
+#ifdef FEATURE_AVX_SUPPORT
+    // When it's a PInvoke call and the call type is USER function, we issue VZEROUPPER here
+	// if the function contains 256bit AVX instructions, this is to avoid AVX-256 to Legacy SSE
+	// transition penalty, assume the user function may contains legacy SSE instruction
+    if (call->IsPInvoke() && call->gtCallType == CT_USER_FUNC && compiler->getFloatingPointInstructionSet() == InstructionSet_AVX)
+    {
+        if (getEmitter()->Contains256bitAVX())
+        {
```

Comment:
Note that if this method is called in a loop, and there are 256-bit AVX instructions emitted after this call, the vzeroupper won't be emitted.